fix(docs): Added docs on providers to quick start guide (#1733)
aws-amplify/amplify-ui: docs/src/pages/getting-started/installation/dependencies.react.mdx

</TabItem>
</Tabs>

After adding the `aws-amplify` and `@aws-amplify/ui-react` dependencies you are now ready to add any of our [components](/components) to your application.

> 

Reviewer: Add a new section here:
```
### Default styling
Load the default styling by importing our CSS:

```jsx
import { AmplifyProvider } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css'; // default theme

export default function App() {
  return (
      <View>
       Your app here
      </View>
  );
};
```

Author: Added

> ### Theming

Reviewer: Maybe this should call out that it's only needed for a custom theme in the title itself?
```suggestion
### Custom Theme
```

Author: Done

> To add [theming](/theming) wrap your application in the `AmplifyProvider` component and include the default styling.

Reviewer: ```suggestion
To add a [custom theme](/theming), wrap your app  in the `AmplifyProvider` component and include the default styling.
```

Author: Updated

<Tabs>

> <TabItem title="Create React App">

Reviewer: I can't comment on the existing code, but it seem weird to name the function App and then have an `<App />` component inside. Maybe put a `<View>` component with "Your app here" string inside.

Author: OK I'll change it to `View`

```
</TabItem>
</Tabs>

> If you're planning on using the [Authenticator](/components/authenticator), also wrap your application in `<Authenticator.Provider/>`.

Reviewer 2: The `Authenticator` is wrapped with a `Provider` internally, so there is no need to wrap an app that uses the `Authenticator` within an `Authenticator.Provider` **unless** the user is attempting to utilize `useAuthenticator` outside the scope of the `Authenticator`, e.g.

```tsx
const ContrivedExample = () => {
  const { user } = useAuthenticator(); // will throw if not wrapped in Authenticator.Provider
  return user.userName ? <p>Hi {user.userName}</p> : <Authenticator />;
}

const App = () => {
  return (
    <AmplifyProvider>
      <Authenticator.Provider>
        <ContrivedExample />
      </Authenticator.Provider>
    </AmplifyProvider>
  )
}
```

Of course the above example is better handled by just moving `ContrivedExample` inside the `Authenticator`:

```tsx
const ContrivedExample = () => {
  const { user } = useAuthenticator(); // works, as it's wrapped in an Authenticator
  return <p>Hi {user.userName}</p>;
}

const App = () => {
  return (
    <AmplifyProvider>
      <Authenticator>
        <ContrivedExample />
      </Authenticator>
    </AmplifyProvider>
  )
}
```

Author: Correct. I hinted at that in the note at the bottom. That's why I think we should include it for everyone using Authenticator, for that scenario, so it's not confusing.

Author: Updated

Reviewer 2: But isn't that directing our users to include superfluous elements in their rendering tree?

Author: OK I moved this around to two different sections, based on our offline discussion

<Tabs>
<TabItem title="Create React App">
```jsx
import { AmplifyProvider, Authenticator } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css'; // default theme

export default function App() {
  return (
    <AmplifyProvider>
      <Authenticator.Provider>
        <App />
      </Authenticator.Provider>
    </AmplifyProvider>
  );
};
```
</TabItem>
<TabItem title="Next.js" >
```jsx
import { AmplifyProvider, Authenticator } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css'; // default theme

export default function App(props) {
  return (
    <AmplifyProvider>
      <Authenticator.Provider>
        <MyApp {...props} />
      </Authenticator.Provider>
    </AmplifyProvider>
  );
};
```
</TabItem>
</Tabs>

> > The `<AmplifyProvider/>` helps direct [theming](/theming) to all the Amplify components in your application. It can optionally be used with a [theme object](/theming#theme-object), to set the look and feel of your app.

Reviewer 2: Unless we are already including angle brackets around component names as a pattern in the docs should we just stick to wrapping in single backticks, e.g. \`AmplifyProvider\`?

Author: I did this because earlier in the guide we use it as the component name. But I can change it back here.

Author: Back to just the component name 5df98d33ee8b28c55e6af2ff7614e30b0198e5d9

>

> > The `<Authenticator.Provider/>` is used with our [Authenticator](/components/authenticator) component. It's guarantees that the [useAuthenticator hook](/components/authenticator#useauthenticator-hook) is available throughout your application. You can see an example of this pattern in the [Protected Routes Guide.](/guides/auth-protected)

Reviewer 3: nit typo: "It guarantees"

Author: Thanks, fixed in #759f2cf

```
</TabItem>
</Tabs>

> The `AmplifyProvider` helps direct [theming](/theming) to all the Amplify components in your application. It can optionally be used with a [theme object](/theming#theme-object), to set the look and feel of your app.

### Authenticator Provider

> If you're planning on using the [Authenticator](/components/authenticator) with the [useAuthenticator hook](/components/authenticator#useauthenticator-hook), for advanced use cases, wrap your application in the `Authenticator.Provider` component.

Reviewer 2: ```suggestion
In advanced use cases where usage of the [`useAuthenticator` hook](/components/authenticator#useauthenticator-hook) outside the scope of the [`Authenticator`](/components/authenticator) is needed, wrap your application inside a `Authenticator.Provider`.
```

Author: OK changed

<Tabs>
<TabItem title="Create React App">
```jsx
import { AmplifyProvider, Authenticator } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css'; // default theme

export default function App() {
  return (
    <Authenticator.Provider>

>       <App />

Reviewer: ```suggestion
      <View>
       Your app here
      </View>
```

Author: Done

    </Authenticator.Provider>
  );
};
```
</TabItem>
<TabItem title="Next.js" >
```jsx
import { AmplifyProvider, Authenticator } from '@aws-amplify/ui-react';
import '@aws-amplify/ui-react/styles.css'; // default theme

export default function App(props) {
  return (
    <Authenticator.Provider>

>       <MyApp {...props} />

Reviewer: Would be nice if this could copy paste into customer apps without any errors from missing `MyApp` component.
```suggestion
      <View>
       Your app here
      </View>
```

Author: Done
